### ISSUE TITLE
Doesn't fetch more than 50 pins

### ISSUE THREAD

**Issue by Korfos** [TakeTwoMerkidentiteit/Pinterest-API#4]
I'm calling getPinsFromBoard with itemsperpage = 100. In the result i have total_items = 50 and items_per_page = 100.

**Comment by dirkgroenen** [comment]
Yes that's a known issue. This API makes use of http://pinterestapi.co.uk/ which doesn't return more than 50 items for some reason.

**Comment by bodyflex** [comment]
why not use the "official" pinterest API that their web client uses as well?

**Comment by dirkgroenen** [comment]
I wasn't aware of that API call, but with that in mind it wouldn't hurt creating a new version of this lib.

**Comment by bodyflex** [comment]
well that endpoint isn't documented, you just have to sniff requests that the Pinterest browser client makes.
the pros are that many endpoints don't require authorization and the data is well-formatted json.
the cons are that there is no certainty of the format of the response since this isn't an official supported API in that sense.

**Comment by dirkgroenen** [comment]
I've been investigating it for a current project and started writing a PHP wrapper for it. At the moment it's available under our [Gitlab](http://gitlab.taketwobeta.nl/dirk/pinterest-php-api) server and only supports fetching pins. We want to extend that one at a later moment and push it to Github when it's finished.

**Comment by bodyflex** [comment]
sounds good! the other endpoints are easy to find with devtools, but simplifying the requests might take some playing around with curl.

**Comment by Korfos** [comment]
I see that this call is limiting the results to 25. Is there a way to specify the count of the returned values? I need 100 latest pins returned.

**Comment by dirkgroenen** [comment]
By setting `$pinterest->itemsperpage = 50;` it should return 50 results. If that doesn't work it might be on the API side. Are you able to retrieve more pins with the repo I shared in the previous comment?

**Comment by Korfos** [comment]
With this pinterest API 50 is the most pins  you can get. Using the call bodyflex mentioned, the call is limited to 25 results. Was wondering if there is a way to make it show more.

**Comment by bodyflex** [comment]
@Korfos let me find out

**Comment by dirkgroenen** [comment]
Ah you were talking about the other API. As Bodyfles is mentioning; there most be a way to paginate between the pins, but this can only be find by experimenting with the requests. 

I might have some time left today, if so I'll check it out ;)

**Comment by dirkgroenen** [comment]
@Korfos @bodyflex just did a little test and it turns out that the only changing parameters in the requests are `bookmark` and `&_`.

Results after paginating: 

```
1st. https://www.pinterest.com/resource/UserPinsResource/get/?source_url=/brunoponzini/pins/&data={"options":{"username":"brunoponzini"},"context":{}}&module_path=App>UserProfilePage>UserInfoBar(tab=pins, spinner=[object Object], resource=UserResource(username=brunoponzini, invite_code=null))&_=1437985406924
2nd. https://www.pinterest.com/resource/UserPinsResource/get/?source_url=/brunoponzini/pins/&data={"options":{"username":"brunoponzini","bookmarks":["LT42MDIzNTcxMzc0Mjk3Nzc4MToyNToyNXxiNDNiNmJmMGQxY2E1OTQ4NTQwOWM5ZTc0Y2QyNmViYmRmZDk2ZTNiODY5OTRlY2Q4ZjA3OTM3NmFkNDdkMjlm"]},"context":{}}&module_path=App>UserProfilePage>UserInfoBar(tab=pins, spinner=[object Object], resource=UserResource(username=brunoponzini, invite_code=null))&_=1437985406927
3rd. https://www.pinterest.com/resource/UserPinsResource/get/?source_url=/brunoponzini/pins/&data={"options":{"username":"brunoponzini","bookmarks":["LT42MDIzNTcxMzc0Mjk0MTQzNzo0OTo1MHw2M2FlMDM3ZTQxNzhmZDIxOTJlN2IwZjFmOGM0YWQ1NmE0Zjk4MDMxNDAyMmQxNGMxYzNlYjQ5ODEwZjFkNTEz"]},"context":{}}&module_path=App>UserProfilePage>UserInfoBar(tab=pins, spinner=[object Object], resource=UserResource(username=brunoponzini, invite_code=null))&_=1437985406936
4th. https://www.pinterest.com/resource/UserPinsResource/get/?source_url=/brunoponzini/pins/&data={"options":{"username":"brunoponzini","bookmarks":["LT42MDIzNTcxMzc0Mjg0NzgyNjo3Mzo3NXw5M2EzOTZkOWE2ZTgyZDA0MDcyOGUwNDNjZTBiNDE2MThlY2YxYmI1Mjk4YTVmNDEyODZmNmE0M2UxNDM2MDc5"]},"context":{}}&module_path=App>UserProfilePage>UserInfoBar(tab=pins, spinner=[object Object], resource=UserResource(username=brunoponzini, invite_code=null))&_=1437985406947
```

`&_` is just a timestamp of the request so that can't be it. Let me get back on this later today.
